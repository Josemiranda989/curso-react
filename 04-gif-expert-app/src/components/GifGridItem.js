import React from 'react'

export const GifGridItem = ({id, title, image}) => {
    
    console.log(image);
  return (
      <div>
          <img src={image} alt={title} />
          <p>{title}</p>
      </div>
  )
}
