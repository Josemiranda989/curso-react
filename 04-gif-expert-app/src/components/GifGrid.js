import React, { useState, useEffect } from "react";
import {GifGridItem} from './GifGridItem'
export const GifGrid = ({ category }) => {

  const [images, setImages] = useState([])

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url =
      "https://api.giphy.com/v1/gifs/search?q=pokemon&limit=10&api_key=XIRrAa3TO5OjXep9tH4PLzHDTfJ7OOxP";
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        image: img.images.downsized_medium.url,
      };
    });

    console.log(gifs);
    setImages(gifs)
  };

  // getGifs()

  return (
    <>
      <h3>{category}</h3>
      
        {images.map((img) => (
          <GifGridItem
            {...img}
            key={img.id}
          />
         ) )}
      
    </>
  );
};
