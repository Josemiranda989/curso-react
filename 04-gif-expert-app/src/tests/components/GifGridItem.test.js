import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("pruebas en GifGridItem.js", () => {
  let wrapper = shallow(<GifGridItem />);

  test("debe de mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
 

});
/* 
    1. Enzyme
    2. Enzyme to JSON
    3. debe de mostrar el componente correctamente
        * shallow
        * wrapper
        * wrapper .toMatchSnapshot()
*/