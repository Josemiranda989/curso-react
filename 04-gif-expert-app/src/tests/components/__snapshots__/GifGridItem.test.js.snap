// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGridItem.js debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <p />
</div>
`;
